pkg/testing/integration/program.go

```diff
@@ -792,7 +794,9 @@ func ProgramTest(t *testing.T, opts *ProgramTestOptions) {
 	prepareProgram(t, opts)
 	pt := newProgramTester(t, opts)
 	err := pt.TestLifeCycleInitAndDestroy()
-	assert.NoError(t, err)
+	if !errors.Is(err, ErrTestFailed) {
```

Comment:
Probably worth a quick comment here that this is so we don't trigger an assert message when we've already failed an assert.